hikari/internal/ux.py

```diff
@@ -198,6 +205,9 @@ def print_banner(package: typing.Optional[str], allow_color: bool, force_color:
         "system_description": " ".join(filtered_system_bits),
     }
 
+    if extra_args:
+        args.update(extra_args)
```

Comment:
Do we really want to allow then to update the values of the existing args?

Reply:
I could switch it out for something like this.
```python
for key in extra_args:
    if key in args:
        raise ValueError("Default $-substitutions cannot be modified")
args.update(extra_args)
```
imo using update is just much cleaner code and fields like `system_description` should be modifiable anyway.

Reply:
I was thinking more on using `.setdefault`, but that approach looks good too.

I am not a fan of allowing people to modify it easily as it could lead to conflics as well as over writing values that third party libs should not be modifying

Reply:
Thats a good point, and also people could accidentally mess something up. I made it so arguments can't be overwritten.